pkg/noderesourcetopology/plugin.go

```diff
@@ -0,0 +1,109 @@
+package noderesourcetopology
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	apiconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+)
+
+type NUMANode struct {
+	NUMAID    int
+	Resources v1.ResourceList
+}
+
+type NUMANodeList []NUMANode
+
+type nodeResTopologyPlugin struct {
+	lister     *listerv1alpha1.NodeResourceTopologyLister
+	namespaces []string
+}
+
+type tmScopeHandler interface {
+	TopologyScopedFilter(pod *v1.Pod, zones topologyv1alpha1.ZoneList) *framework.Status
+	TopologyScopedScore(pod *v1.Pod, zones topologyv1alpha1.ZoneList, scorerFn scoreStrategy, resourceToWeightMap resourceToWeightMap) (int64, *framework.Status)
+}
+
+type podScopeHandler struct{}
```

Comment:
An empty struct doesn't look necessary to me. Can we switch the interface to a struct, with two functions as fields? Something like https://gist.github.com/Huang-Wei/16daaf5352b1bc576de653c3cc660f29.